fixtures/provider.py

```diff
@@ -67,8 +76,26 @@ def _artifactor_skip_providers(request, providers, skip_msg):
     pytest.skip(skip_msg)
 
 
-def _setup_provider_verbose(provider):
+def _setup_provider_verbose(request, provider):
     try:
+        if request.config.option.provider_limit > 0:
+            existing_providers = [
+                p for p in list_providers(use_global_filters=False)
+                if p.exists and p.key != provider.key]
```

Comment:
I think ideally we would want to get current appliance in here through the `appliance` fixture and use `appliance.managed_known_providers` but all that might still change once we get collections so... this is good enough.